Sources/Commands/SwiftTool.swift

```diff
@@ -489,22 +488,65 @@ public class SwiftTool {
 
     }
 
-    private func editsDirectory() throws -> AbsolutePath {
+    /// Returns the currently active workspace.
+    func getActiveWorkspace() throws -> Workspace {
+        if let workspace = _workspace {
+            return workspace
+        }
+
+        // we do not take an observabilityScope in the workspace initializer as the workspace is designed to be long lived
+        // instead observabilityScope is passed into the individual workspace methods which are short lived
+        // however, the workspace initializer may need to emit diagnostics, for which the CLI sets this thread local
+        Thread.current.threadDictionary["observabilityScope"] = observabilityScope
+
+        let isXcodeBuildSystemEnabled = self.options.buildSystem == .xcode
+        let repositoryProvider = GitRepositoryProvider(processSet: self.processSet)
+        let delegate = ToolWorkspaceDelegate(self.outputStream, logLevel: self.logLevel, observabilityScope: self.observabilityScope)
+        let workspace = try Workspace(
+            fileSystem: localFileSystem,
+            location: .init(
+                workingDirectory: self.buildPath,
+                editsDirectory: self.getEditsDirectory(),
+                resolvedVersionsFile: self.getResolvedVersionsFile(),
+                localConfigurationDirectory: try self.getLocalConfigurationDirectory(),
+                sharedConfigurationDirectory: self.sharedConfigurationDirectory,
+                sharedSecurityDirectory: self.sharedSecurityDirectory,
+                sharedCacheDirectory: self.sharedCacheDirectory
+            ),
+            authorizationProvider: self.getAuthorizationProvider(),
+            resolutionConfiguration: .init(
+                updateEnabled: !options.skipDependencyUpdate,
+                prefetchingEnabled: options.shouldEnableResolverPrefetching,
+                additionalFileRules: isXcodeBuildSystemEnabled ? FileRuleDescription.xcbuildFileTypes : FileRuleDescription.swiftpmFileTypes,
+                sharedRepositoriesCacheEnabled: self.options.useRepositoriesCache,
+                fingerprintCheckingMode: self.options.resolverFingerprintCheckingMode
+            ),
+            customManifestLoader: self.getManifestLoader(), // FIXME: ideally we would not customize the manifest loader
```

Comment:
Since different interfaces to libSwiftPM might well want to have their own manifest loaders, I think it does make sense to customize it here.  I view SwiftTools as a peer client of any IDE that uses libSwiftPM, and if they customize it, it makes sense for SwiftTools to do it too.

Reply:
I find it surprising that SwiftTool should need to customize it. For IDEs, this is a good extension point to inject a loader with additional capabilities like plugins, but SwiftTool is part of the same codebase so it surprising the default loader does not work out of the box. iow what is the value of the default loader if it is not used anywhere?